ts/Checkbox/index.tsx

```diff
@@ -10,32 +11,50 @@ const defaultProps = {
   indeterminate: false
 };
 
-/**
- */
-export default class Checkbox extends MaterialComponent {
-  constructor() {
-    super();
-    this.componentName = 'checkbox';
-    this._mdcProps = ['disabled'];
-  }
-  componentDidMount() {
-    this.MDComponent = new MDCCheckbox(this.control);
-    toggleCheckbox(defaultProps, this.props, this.MDComponent);
+export interface ICheckboxProps {
+  indeterminate?: boolean;
+  disabled?: boolean;
+  onChange?: JSX.GenericEventHandler;
```

Comment:
Question why is this `JSX.GenericEventHandler` and someplaces we've used (e: event) => void.
Also is this really necessary? cuz i dont see `onClick` for button?

Reply:
So which one should we adopt everywhere?

Reply:
I think we should go with `JSX.GenericEventHandler` as preact defines and uses that one